### PR TITLE
Implicitly initialize @onready variables

### LINKED ISSUE #52223: onready variable treated as _ready method
### Godot version

4d08a73

### System information

W10

### Issue description

Variation of #42506

`_ready()` from base class is not called when there is an `onready` variable in the inheriting class.

### Steps to reproduce

1. Run attached project
2. See that "ready" is not printed
3. Comment out `onready` variable
4. Run
5. "ready" now appears

### Minimal reproduction project

[ReproductionProject.zip](https://github.com/godotengine/godot/files/7072634/ReproductionProject.zip)

EDIT:
This also works the other way around. Any `onready` variable in base class is not initialized when you have `_ready()` method in your inherited class, unless you call `super._ready()`.

### PR DESCRIPTION
**Pull request by KoBeWi**
Fixes #52223

No idea what I am doing. I vaguely remember vnen mentioning this as potential fix, so I tried doing it and accidentally it works, so cool I guess?

### CODE REVIEW

File: scene/main/node.cpp [hunk 1]
```diff
@@ -152,6 +152,9 @@ void Node::_notification(int p_notification) {
 			}
 
 			GDVIRTUAL_CALL(_ready);
+			if (has_method(CoreStringNames::get_singleton()->implicit_ready)) {
+				call(CoreStringNames::get_singleton()->implicit_ready);
+			}
```
Comment (KoBeWi) [inline scene/main/node.cpp:157]: Wasn't sure where and how to call this.

**Comment by sairam4123** [comment]
Seems like you have notified someone accidentally.

**Comment by vnen** [comment]
The idea I had is a bit different. I think the `@implicit_ready` should always exist and always call the parent's `@implicit_ready`. Then the `_ready` function should call `@implicit_ready` before anything. The only issue here is to not add a `_ready` function when it's not needed since it will be called just to do nothing, so it needs a mechanism to detect that in the inheritance tree.

Another issue is that the implicit one will be shown in the methods list. It is a problem with `@implicit_new` too, but I want to fix it by not actually adding it to the list. I also don't think the name needs to be in CoreStringNames for instance.

Let me know if you want to tackle this yourself, I can give some more advice. Otherwise I'll probably get to this soon.

**Comment by KoBeWi** [comment]
> Another issue is that the implicit one will be shown in the methods list.

IMO the list should just filter out methods that start with `@`.
Not sure which list you mean though, I only know the problem exists in AnimationPlayer.

> I also don't think the name needs to be in CoreStringNames for instance.

Probably. I used the name a few times in multiple files, so just made it a constant.

> Let me know if you want to tackle this yourself, I can give some more advice. Otherwise I'll probably get to this soon.

This PR was some hard improvisation. I think it would be faster if you handled this, as you have better idea what to do and how.

**Comment by KoBeWi** [comment]
Another thing worth to keep in mind is #33620. IMO `_ready()` and `@onready` shouldn't be tied. Any onready variable should be initialized before ready is called. This PR ensures that and it would be nice if the superseding one did this too.

**Comment by vnen** [comment]
> Another thing worth to keep in mind is #33620. IMO `_ready()` and `@onready` shouldn't be tied. Any onready variable should be initialized before ready is called. This PR ensures that and it would be nice if the superseding one did this too.

`@onready` variables acts as if they are inserted in the beginning of the `_ready()` method (whether `_ready()` is defined or not). The only difference is that `@onready` should always be called across the inheritance tree, so they should be called even without an explicit `super` call.

**Comment by vnen** [comment]
> So will your fix resolve #33620? Even if it would need explicit `super()` call?

It should, yes. The issue seems to be related to the order of multilevel calls, which we don't have anymore.

**Comment by vnen** [comment]
I'm closing this in favor of my own PR: #62255

Thanks for bringing this up!

**Pull request closed without merging**